go.mod

```diff
@@ -77,42 +77,12 @@ replace k8s.io/client-go => k8s.io/client-go v0.20.4
 
 replace k8s.io/api => k8s.io/api v0.20.4
 
-// >v1.2.0 has some conflict with prometheus/alertmanager. Hence prevent the upgrade till it's fixed.
-replace github.com/satori/go.uuid => github.com/satori/go.uuid v1.2.0
-
 // Use fork of gocql that has gokit logs and Prometheus metrics.
 replace github.com/gocql/gocql => github.com/grafana/gocql v0.0.0-20200605141915-ba5dc39ece85
 
 // Using a 3rd-party branch for custom dialer - see https://github.com/bradfitz/gomemcache/pull/86
 replace github.com/bradfitz/gomemcache => github.com/themihai/gomemcache v0.0.0-20180902122335-24332e2d58ab
 
-// Pin github.com/go-openapi versions to match Prometheus alertmanager to avoid
-// breaking changing affecting the alertmanager.
-replace github.com/go-openapi/errors => github.com/go-openapi/errors v0.19.4
-
-replace github.com/go-openapi/loads => github.com/go-openapi/loads v0.19.5
-
-replace github.com/go-openapi/runtime => github.com/go-openapi/runtime v0.19.15
-
-replace github.com/go-openapi/spec => github.com/go-openapi/spec v0.19.8
-
-replace github.com/go-openapi/strfmt => github.com/go-openapi/strfmt v0.19.5
-
-replace github.com/go-openapi/swag => github.com/go-openapi/swag v0.19.9
-
-replace github.com/go-openapi/validate => github.com/go-openapi/validate v0.19.8
-
-// Pin these, which are updated as dependencies in Prometheus; we will take those updates separately and carefully
-replace (
-	github.com/aws/aws-sdk-go => github.com/aws/aws-sdk-go v1.37.8
-	github.com/google/pprof => github.com/google/pprof v0.0.0-20210208152844-1612e9be7af6
-	github.com/miekg/dns => github.com/miekg/dns v1.1.38
-	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.9.0
-	golang.org/x/crypto => golang.org/x/crypto v0.0.0-20201208171446-5f87f3452ae9
-	golang.org/x/net => golang.org/x/net v0.0.0-20210119194325-5f4716e94777
-	golang.org/x/oauth2 => golang.org/x/oauth2 v0.0.0-20210210192628-66670185b0cd
-	golang.org/x/sync => golang.org/x/sync v0.0.0-20201207232520-09787c993a3a
-	golang.org/x/sys => golang.org/x/sys v0.0.0-20210124154548-22da62e12c0c
-	google.golang.org/api => google.golang.org/api v0.39.0
-	google.golang.org/grpc => google.golang.org/grpc v1.34.0
-)
+// Need to upgrade Prometheus and Alertmanager in Thanos too, due to some breaking changes.
+// We use a commit from a branch until https://github.com/thanos-io/thanos/pull/4094 is not merged.
+replace github.com/thanos-io/thanos v0.13.1-0.20210401085038-d7dff0c84d17 => github.com/thanos-io/thanos v0.19.1-0.20210422104148-b9144ea1b912
```

Comment:
My plan is to keep this, so that we'll not block this PR by Thanos. I will remove this replace once Thanos PR is merged.